File: kmeans/kmeans.py
from sklearn.cluster import KMeans
import matplotlib.pyplot as plt
from sklearn.datasets import make_blobs

# Create sample data
X, _ = make_blobs(n_samples=200, centers=3, random_state=42)

# Apply KMeans
kmeans = KMeans(n_clusters=3, random_state=42)
kmeans.fit(X)

# Plot clusters
plt.scatter(X[:,0], X[:,1], c=kmeans.labels_)
plt.scatter(kmeans.cluster_centers_[:,0], kmeans.cluster_centers_[:,1], color='red', marker='X')
plt.show()

File: kmeans/kmeans.R
# Sample data
set.seed(42)
data <- matrix(rnorm(200), ncol=2)

# Kmeans with 3 clusters
result <- kmeans(data, 3)

# Plot
plot(data, col=result$cluster)
points(result$centers, col=1:3, pch=8, cex=2)
